File: Python/605. Can Place Flowers.py
from typing import List


class Solution:
    # noinspection PyMethodMayBeStatic
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True

        length = len(flowerbed)

        i = 0
        while i < length:
            if flowerbed[i] == 1:
                i += 2
            elif i + 1 >= length or flowerbed[i + 1] == 0:
                n -= 1
                i += 2
            else:
                i += 3

            if n == 0:
                return True

        return False

File: PHP/605. Can Place Flowers.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $flowerbed
     */
    public function canPlaceFlowers(array $flowerbed, int $n): bool
    {
        if ($n === 0) {
            return true;
        }

        $length = count($flowerbed);
        $i = 0;

        while ($i < $length) {
            if ($flowerbed[$i] == 1) {
                $i += 2;
            } elseif ($flowerbed[$i + 1] !== 1) {
                $n--;
                $i += 2;
            } else {
                $i += 3;
            }

            if ($n === 0) {
                return true;
            }
        }

        return false;
    }
}
